### PR TITLE
feat(): Add additional input validation regex and tests

### PR DESCRIPTION
**Pull request by douglasg14b**
- Add regex validation for the `metadata.name` and `dependencies` properties for the key manifest
- Add tests to test for these manifest requirements
- Add additional `sed` to replace `/` in git branch name with `-` in github workflows

### CODE REVIEW

File: tweek.code-workspace [hunk 1]
```diff
@@ -21,7 +21,6 @@
 		}
 	],
 	"settings": {
-		"workbench.editor.enablePreview": false,
```
Comment (rafikul-sekh) [inline tweek.code-workspace:24]: Just curious, why we need this change?

**Review by rafikul-sekh — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -27,7 +27,7 @@
     "format:go": "gofmt -l -w"
   },
   "devDependencies": {
-    "husky": "^4.3.8",
+    "husky": "^8.0.1",
```
Comment (ahardewig) [inline package.json:30]: No code changes need for 4 major versions?

**Review by ahardewig — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (douglasg14b) [reply package.json:30]: Not from my understanding, nope. it's just a pre-commit hook, and older versions of husky have issues actually running on WSL (Another repo also had this issue for me).

File: tweek.code-workspace (hunk 1 above)
Comment (rafikul-sekh) [reply tweek.code-workspace:24]: Sounds good! This should be fine.

File: package.json (hunk 2 above)
Comment (AleF83) [reply package.json:30]: There are breaking changes.
https://github.com/typicode/husky-4-to-8
Don't have to use this script. I just googled it. But the way that husky manages the hooks has been changed.
They were stored as json or yaml and now they're stored in `.husky` folder as bash scripts

Comment (douglasg14b) [reply package.json:30]: Ah gotcha, will investigate

**Review by AleF83 — COMMENTED** [review]
Please bump the version of tweek-authoring in `/services/authoring/package.json`

**Pull request merged**